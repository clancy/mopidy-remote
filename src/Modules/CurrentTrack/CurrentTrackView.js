/* @flow */
import React, { Component, PropTypes } from 'react';
import Controls from './Controls'
import Progress from './Progress'
import TrackInfo from './TrackInfo'
import AlbumArt from './AlbumArt'
import * as MopidyActions from '../Mopidy/Actions'
import {
  View,
  StyleSheet
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons'

const CurrentTrackView = React.createClass({
  propTypes: {
    length: PropTypes.number,
    position: PropTypes.number.isRequired,
    playing: PropTypes.bool.isRequired,
    shuffleEnabled: PropTypes.bool.isRequired,
    repeatEnabled: PropTypes.bool.isRequired,
    inLibrary: PropTypes.bool.isRequired,
    trackName: PropTypes.string,
    artists: PropTypes.arrayOf(React.PropTypes.shape({
      name: PropTypes.string.isRequired
    })),
    albumArtUri: PropTypes.string
  },
  playPause() {
    if(this.props.playing){
      this.props.dispatch(MopidyActions.pause());
    } else {
      this.props.dispatch(MopidyActions.play());
    }
  },
  previous() {
    this.props.dispatch(MopidyActions.previousTrack());
  },
  next() {
    this.props.dispatch(MopidyActions.nextTrack());
  },
  shuffle() {
    this.props.dispatch(MopidyActions.shuffle(!this.props.shuffleEnabled));
  },
  repeat() {
    this.props.dispatch(MopidyActions.repeat(!this.props.repeatEnabled));
  },
  render() {
    return (
      <View style={styles.container}>
        <AlbumArt albumArtUri={this.props.albumArtUri}/>
        <TrackInfo inLibrary={this.props.inLibrary} trackName={this.props.trackName} artists={this.props.artists}/>
        <Progress playing={this.props.playing} length={this.props.length} position={this.props.position}/>
        <Controls
          previous={this.previous}
          next={this.next}
          shuffle={this.shuffle}
          repeat={this.repeat}
          playPause={this.playPause}
          playing={this.props.playing}
          shuffleEnabled={this.props.shuffleEnabled}
          repeatEnabled={this.props.repeatEnabled} />
      </View>
    );
  }
});

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})

export default CurrentTrackView;
