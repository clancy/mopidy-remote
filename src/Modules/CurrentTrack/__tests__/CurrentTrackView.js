import 'react-native';
import React from 'react';
import CurrentTrackView from '../CurrentTrackView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('CurrentTrackView tests', () => {
  test('renders CurrentTrackView correctly', () => {
    const tree = renderer.create(
      <CurrentTrackView
        length={0}
        position={0}
        playing={false}
        shuffleEnabled={false}
        repeatEnabled={false}
        inLibrary={false}
        trackName={'trackname'}
        artists={[{name:'asd'}, {name:'fds'}]}
        albumArtUri= {'http://album.art/uri'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
