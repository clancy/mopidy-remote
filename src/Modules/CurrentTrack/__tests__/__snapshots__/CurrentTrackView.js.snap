exports[`CurrentTrackView tests renders CurrentTrackView correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "flex": 3,
        },
        null,
      ]
    }>
    <Image
      source={
        Object {
          "uri": "http://album.art/uri",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      } />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }>
    <View
      style={
        Object {
          "flex": 15,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 35,
            },
            Object {
              "color": "white",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 70,
          "flexDirection": "column",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }>
        trackname
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
        asd, fds
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 15,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 35,
            },
            Object {
              "color": "white",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }>
    <View
      style={
        Object {
          "flex": 15,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
        0:00
      </Text>
    </View>
    <RCTProgressView
      progress={0}
      progressTintColor="green"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "flex": 70,
          },
        ]
      } />
    <View
      style={
        Object {
          "flex": 15,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
        0:00
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 25,
            },
            Object {
              "color": "white",
              "marginLeft": 10,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 35,
            },
            Object {
              "color": "white",
              "marginLeft": 10,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 35,
          "borderWidth": 0.5,
          "height": 70,
          "justifyContent": "center",
          "width": 70,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 50,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "white",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }>
          
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 35,
            },
            Object {
              "color": "white",
              "marginLeft": 10,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 25,
            },
            Object {
              "color": "white",
              "marginLeft": 10,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </View>
</View>
`;
